src/pages/login-page/LoginPage.tsx

```diff
@@ -67,15 +67,15 @@ const LoginPage = () => {
         getJobStatus(Number(jobId)).then(async ({ status }) => {
           const creds = JSON.parse(localStorage.getItem(constants.CREDENTIALS)!);
           const metamaskService = new MetaMaskService();
-          const userDataString = await metamaskService.decryptMessage(creds, account);
-          const userData: EncryptedCredentials = JSON.parse(userDataString);
-
-          dispatch(setUserData(userData));
 
           if (status === 'active') {
             socket.on(
               'signUpSuccessful',
               async ({ token, address, alias }: SignUpSuccessfulEventResponse) => {
+                const userDataString = await metamaskService.decryptMessage(creds, account);
```

Comment:
There is code duplied, but works, It's already tested!
You can add a TODO to refactor